README.md

```diff
@@ -2142,6 +2237,42 @@ authClient.tokenManager.off('renewed');
 authClient.tokenManager.off('renewed', myRenewedCallback);
 ```
 
+### `authStateManager`
+
+`AuthStateManager` evaluates and emits `AuthState` based on the events from `TokenManager` for downstream clients to consume.
+
+The emitted `AuthState` object includes:
+
+* `isPending`: true in the time after page load (first render) but before the asynchronous methods to see if the tokenManager is aware of a current authentication.
+* `isAuthenticated`: true if the user is considered authenticated. Normally this is true if both an idToken and an accessToken are present in the tokenManager, but this behavior can be overridden if you passed an isAuthenticated callback in the [configuration](#configuration-reference).
+* `accessToken`: the JWT accessToken for the currently authenticated user (if provided by the scopes).
+* `idToken`: the JWT idToken for the currently authenticated user (if provided by the scopes).
+* `error`: contains the error returned if an error occurs in the `authState` evaluation process.
+
+Subscribes to `authStateChange` event:
+
+```javascript
+authClient.authStateManager.onAuthStateChange((authState) => {
+  // handle the latest evaluated authState, like integrate with client framework's state management store
+});
+```
+
+#### `authStateManager.getAuthState()`
+
+Gets latest evaluated `authState` from the `authStateManager`. The `authState` is re-evaluated when .authStateManager.updateAuthState() is called.
+
```

Comment:
Can mention that `updateAuthState` should always produce a unique object, so the result of `getAuthState` can be compared by identity